pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -50,15 +57,76 @@ func New(cfg *setting.Cfg, validator models.PluginRequestValidator, tracer traci
 				return
 			}
 			datasourceLabelName, err := metricutil.SanitizeLabelName(datasourceName)
-
 			if err != nil {
 				return
 			}
+
+			_, enableSocksProxy := opts.CustomOptions["socks_proxy"]
+			if cfg.IsFeatureToggleEnabled(featuremgmt.FlagSecureSocksDatasourceProxy) && enableSocksProxy {
+				err = newSecureSocksProxy(cfg, transport)
+				if err != nil {
+					logger.Error("Failed to enable the socks proxy", "error", err.Error(), "datasource", datasourceName)
+				}
+			}
+
 			newConntrackRoundTripper(datasourceLabelName, transport)
 		},
 	})
 }
 
+// newSecureSocksProxy takes a http.DefaultTransport and wraps it in a socks5 proxy with TLS
+func newSecureSocksProxy(cfg *setting.Cfg, transport *http.Transport) error {
+	// all fields must be specified to use the proxy
+	if cfg.SecureSocksDSProxy.RootCA == "" {
+		return errors.New("missing rootCA")
+	} else if cfg.SecureSocksDSProxy.ClientCert == "" || cfg.SecureSocksDSProxy.ClientKey == "" {
+		return errors.New("missing client key pair")
+	} else if cfg.SecureSocksDSProxy.ServerName == "" {
+		return errors.New("missing server name")
+	} else if cfg.SecureSocksDSProxy.ProxyAddress == "" {
+		return errors.New("missing proxy address")
+	}
+
+	certPool := x509.NewCertPool()
+	for _, rootCAFile := range strings.Split(cfg.SecureSocksDSProxy.RootCA, " ") {
+		// nolint:gosec
+		// The gosec G304 warning can be ignored because `rootCAFile` comes from config ini.
+		pem, err := os.ReadFile(rootCAFile)
```

Comment:
Shall we provide the actual keys/certs instead of providing the paths ? 
Similar to https://github.com/grafana/grafana/blob/a45ef61d255ff463fe3e70edb50b73c8f1b3afcd/pkg/services/provisioning/datasources/testdata/all-properties/all-properties.yaml#L28-L30

Reply:
Adding this here for future clarification (since we got to discuss this over Google Meets), but we decided to keep it as a file since that is what the precedent is for tls information defined in the config.ini (for example in the http_proxy as well as for auth configurations). The times where it hasn't been a file were times when it is defined in the datasource json instead